lib/chai/core/assertions.js

```diff
@@ -1402,7 +1402,8 @@ module.exports = function (chai, _) {
       }
     }
 
-    if (errMsgMatcher) {
+    if (errMsgMatcher !== undefined && errMsgMatcher !== null &&
+        caughtErr !== undefined && caughtErr !== null) {
```

Comment:
The way you have it now is fine. It's a stylistic preference. Only mentioned it cause I was already commenting anyway on the `caughtErr` part. :D